google-http-client/src/main/java/com/google/api/client/json/webtoken/JsonWebToken.java

```diff
@@ -361,6 +362,12 @@ public String toString() {
     return Objects.toStringHelper(this).add("header", header).add("payload", payload).toString();
   }
 
+  public String getRawToken() {
+    return Base64.encodeBase64URLSafeString(header.toString().getBytes())
+        + "."
```

Comment:
Is there supposed to be a signature here too? Or can there be?

Reply:
nope, the `JsonWebToken` class doens't have a place for the signature (it only accepts the `Heaer` and `Payload`; the `JsonWebSignature` one extends the base to tack on the signature bit

Reply:
I'm trying to work my way through RFC 7519. Is it possible for there to be a JWT that is neither a JWE or a JWS? Section seems to suggest that this is possible, but that the signature in this case is the empty string and there should be a terminating period.

Reply:
oh, you mean if `JsonWebToken.getRawToken()` should emit a `foo.bar.`  and the only thinkg `JsonWebSignature.getRawToken()` does is direct appending (w/o a '.' prefix)?  good question, i don't know that much detail in the specs...(IMO i doubt it but i gotta do some reading tomorrwo)

Reply:
ok, AFAIK, the trailing `.` isn't needed at all in the output of `JsonWebToken.class` ..its just a delimiter...  infact, i'm not sure where i'd use just the output of `JsonWebToken.getRawToken` standalone ouside of this context ..i'm guessing that was added incase we add on a `JsonWebEncryption`... i'd only be interested in `JWS`  or maybe `JWE`